File: keiner_lindarte_aro/instructor_henry/mis_programas/practicas/dart/practicas_3/puedes_pasar_o_no.dart
import 'dart:io';

void main () {

  print('----------------------------------------------------------');

  print('!!Bienvenido a la licoleria!!');

  print('----------------------------------------------------------');

  print(' ');

  print('----------------------------------------------------------');

  print('Requisitos para entrar:');
  
  print('----------------------------------------------------------');

  print('1- Ser mayor de edad (18)');
  print('2- Tener el DNI a la mano');

  print('----------------------------------------------------------');

  print('Cuantas personas van a entrar:');
  int cuantos = int.parse(stdin.readLineSync()!);

  for (int vueltas = 1; vueltas <= cuantos; vueltas++) {

    print('Cuantos años tienes:');
    int age = int.parse(stdin.readLineSync()!);

    print('Traes tu DNI:');
    String dni = stdin.readLineSync()!;

    print('----------------------------------------------------------');

    if (age < 18 && dni == 'no') {
      print('Usted no puede entrar por menor de edad joven');
    } else if (age >= 18 && dni == 'si') {
        print('Usted puede entrar');
      } else if (age >= 18 && dni == 'no') {
        print('No se puede entrar, ya que no trajo su DNI');
      } else if (age < 18 && dni == 'si') {
        print('Acaso no sabe que este lugar es para mayores de edad joven');
      }

    print('----------------------------------------------------------');

  }

}

File: keiner_lindarte_aro/instructor_henry/mis_programas/practicas/ruby/practicas_3/puedes_pasar_o_no.rb

puts "----------------------------------------------------------"

puts "!!Bienvenido a la licoleria!!"

puts "----------------------------------------------------------"

puts " "

puts "----------------------------------------------------------"

puts "Requisitos para entrar:"

puts "----------------------------------------------------------"

puts "1- Ser mayor de edad (18)"
puts "2- Tener el DNI a la mano"

puts "----------------------------------------------------------"

puts "Cuantas personas van a entrar:"
cuantos = gets.chomp.to_i

for vueltas in 1..cuantos

    puts "----------------------------------------------------------"

    puts "Cuantos años tienes:"
    age = gets.chomp.to_i

    puts "Traes tu DNI:"
    dni = gets.chomp

    puts "----------------------------------------------------------"

    if age < 18 && dni == "no"
        puts "Usted no puede entrar por menor de edad joven"
    elsif age >= 18 && dni == "si"
        puts "Usted puede entrar"
    elsif age >= 18 && dni == "no"
            puts "No se puede entrar, ya que no trajo su DNI"
    elsif age < 18 && dni == "si"
            puts "Acaso no sabe que este lugar es para mayores de edad joven"
    end

    puts "----------------------------------------------------------"

end